File: FW 4.0/ESystem/_Extensions/ExceptionExt.vb
Option Explicit On
Option Strict On

Namespace Extensions

  Public Module ExceptionExt
    ''' <summary>
    ''' Returns joined message from exception and all inner exceptions.
    ''' </summary>
    ''' <param name="exception"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    <System.Runtime.CompilerServices.Extension()> _
  Public Function Info(ByVal exception As Exception) As String
      Dim ret As New Text.StringBuilder()

      Dim ex As Exception = exception
      While ex IsNot Nothing
        ret.Append(ex.Message)
        ret.Append(" --> ")
        ex = ex.InnerException
      End While

      Return ret.ToString()
    End Function
  End Module

End Namespace

File: _Previous/Backup/SVL 4/ESystem/_Extensions/ExceptionExt.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace ESystem.Extensions
{
  public static class ExceptionExt
  {
    /// <summary>
    /// Returns joined message from exception and all inner exceptions.
    /// </summary>
    /// <param name="exception"></param>
    /// <returns></returns>
    /// <remarks></remarks>
    [Obsolete("Use GetMessages() extension method instead.")]
    public static string Info(this Exception exception)
    {
      string ret = exception.GetMessages();
      return ret;
    }

    /// <summary>
    /// Returns joined message from exception and all inner exceptions.
    /// </summary>
    /// <param name="exception"></param>
    /// <returns></returns>
    /// <remarks></remarks>
    public static string GetMessages(this Exception exception)
    {
      StringBuilder ret = new StringBuilder();

      Exception ex = exception;
      while (ex != null)
      {
        ret.Append(ex.Message);
        ret.Append(" --> ");
        ex = ex.InnerException;
      }

      return ret.ToString();
    }

    /// <summary>
    /// Returns joined ToString() sets from exception and all inner exceptions.
    /// </summary>
    /// <param name="exception"></param>
    /// <returns></returns>
    /// <remarks></remarks>
    public static string GetToStrings(this Exception exception)
    {
      StringBuilder ret = new StringBuilder();

      Exception ex = exception;
      while (ex != null)
      {
        ret.Append(ex.ToString());
        ret.Append(" --> ");
        ex = ex.InnerException;
      }

      return ret.ToString();
    }
  }

}
